base/lock.jl

```diff
@@ -9,6 +18,48 @@ type ReentrantLock
     ReentrantLock() = new(nothing, Condition(), 0)
 end
 
+"""
+    islocked(the_lock) -> Status (Boolean)
+
+Check whether the lock is held by any task/thread.
+This should not be used for syncronization (see instead `trylock`).
+"""
+function islocked(rl::ReentrantLock)
+    return rl.reentrancy_cnt != 0
+end
+
+"""
+    trylock(the_lock) -> Success (Boolean)
+
+Acquires the lock if it is available,
+returning `true` if successful.
+If the lock is already locked by a different task/thread,
+return `false`.
+
+Each successful "trylock" must be matched by an "unlock"
+"""
+function trylock(rl::ReentrantLock)
+    t = current_task()
+    if rl.reentrancy_cnt == 0
+        rl.locked_by = t
```

Comment:
it's not that mucg more verbose. as is, you have to go read the type declaration to know if this field is nullable, it's very nonlocal and makes reading the code substantially more confusing. would rather have easy to understand code that's a few characters longer.